### ISSUE TITLE
UseNotifyAllInsteadOfNotify detect false-positive in the case: foo().notify(arg1,arg2...)

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2442]
**Affects PMD Version:** 
6.22.0

**Rule:**
UseNotifyAllInsteadOfNotify

**Description:**
UseNotifyAllInsteadOfNotify detect false-positive in the case: foo().notify(arg1,arg2...)
This rule is implemented through xpath search:
```
//StatementExpression/PrimaryExpression
[PrimarySuffix/Arguments[@Size = '0']]
[
    PrimaryPrefix[
        ./Name[@Image='notify' or ends-with(@Image,'.notify')]
        or ../PrimarySuffix/@Image='notify'
        or (./AllocationExpression and ../PrimarySuffix[@Image='notify'])
    ]
]
```
The AST analyse result for the code sample: 
```
<PrimaryPrefix BeginColumn='17' BeginLine='131' EndColumn='27' EndLine='131' FindBoundary='false' Image='' SingleLine='true' SuperModifier='false' ThisModifier='false'>
    <Name BeginColumn='17' BeginLine='131' EndColumn='27' EndLine='131' FindBoundary='false' Image='getInjector' SingleLine='true' />
</PrimaryPrefix>
<PrimarySuffix ArgumentCount='0' Arguments='true' ArrayDereference='false' BeginColumn='28' BeginLine='131' EndColumn='29' EndLine='131' FindBoundary='false' Image='' SingleLine='true'>
    <Arguments ArgumentCount='0' BeginColumn='28' BeginLine='131' EndColumn='29' EndLine='131' FindBoundary='false' Image='' SingleLine='true' Size='0' />
</PrimarySuffix>
<PrimarySuffix ArgumentCount='-1' Arguments='false' ArrayDereference='false' BeginColumn='30' BeginLine='131' EndColumn='36' EndLine='131' FindBoundary='false' Image='notify' SingleLine='true' />
<PrimarySuffix ArgumentCount='3' Arguments='true' ArrayDereference='false' BeginColumn='37' BeginLine='131' EndColumn='80' EndLine='131' FindBoundary='false' Image='' SingleLine='true'>
    <Arguments ArgumentCount='3' BeginColumn='37' BeginLine='131' EndColumn='80' EndLine='131' FindBoundary='false' Image='' SingleLine='true' Size='3'>
```
<code>PrimarySuffix/Arguments[@Size = '0']</code> match <code>getInjector()</code>, and then <code>../PrimarySuffix/@Image='notify'</code> match <code>notify(child.getCanonicalName(), rejection, cause);</code>

**Code Sample demonstrating the issue:**
```
getInjector().notify(child.getCanonicalName(), rejection, cause);
```

**Expected outcome:**
false-positive

**Running PMD through:**
CLI
